### ISSUE TITLE
Passing references ResourceList and ResourceItem

### ISSUE THREAD

**Issue by resistorsoftware** [Shopify/polaris-react#3005]
I have a **ResourceList** component that renders **ResourceItem** components with a callback. 

When the last **item** is rendered, I would like a callback to receive the item so that I can run some logic. Specifically, if the item is visible on the screen. I understand I can use refs to manage values that persist after each render. So I am thinking this is an ideal way to manage my needs. 

However if I try and add a ref={someFunction} in my renderItem callback, where items are rendered, React throws up an error telling me I am doing it wrong, and I cannot use refs here. 

So I am a little stumped. How can I pass a ref to the item to my HOC callback, so that I do a data fetch when the item is scrolled on screen? If my HOC has a callback lastProduct(), how can I call it? The following does not work. 

    <ResourceItem
        id={id}
        title={title}
        productType={productType}
        imageSrc={imageSrc}
        ref={index === data.products.length - 1 ? lastProduct : null}
      >

**Comment by resistorsoftware** [comment]
Never mind I figured it out.

**Comment by Webkadabra** [comment]
> Never mind I figured it out.

do you mind sharing how you did it?

**Comment by resistorsoftware** [comment]
I have no idea, that was in 2020. I gave up on the Shopify React Polaris a long time ago, and now I just use the Polaris library through view components. 

I remember a kind soul at Shopify helped me with this, as all I wanted was infinite scroll instead of the crappy paging mechanism for lists. They sent me a codepen that should've worked, but like I say, I walked away long ago from React, so you're out of luck as I have no desire to search my 2020 codes for anything.
